src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -275,7 +267,14 @@ public static string GetFileHash(string path)
                 matchingExistingInstaller.Platform = newInstaller.Platform;
             }
 
-            installerMismatch = unmatchedInstallers.Any() || multipleMatchedInstallers.Any();
+            if (installerMismatch = unmatchedInstallers.Any() || multipleMatchedInstallers.Any())
+            {
+                List<Exception> exceptionList = new List<Exception>();
+                unmatchedInstallers.ForEach(i => exceptionList.Add(new InstallerExceptions.UnmatchedInstallerException(i)));
+                multipleMatchedInstallers.ForEach(i => exceptionList.Add(new InstallerExceptions.MultipleMatchedInstallerException(i)));
+                throw new AggregateException(exceptionList);
```

Comment:
Instead of an aggregate exception, what about an InstallerMatchException or something like that, with two properties... List<Installer> MultipleMatchedInstallers and List<Installer> UnmatchedInstallers?

That seems more concise to me, but tell me if you disagree

Reply:
totally agree. I created an InstallerMatchException that does just that. should be cleaner now, thanks for the suggestion.